### PR TITLE
home work week 2

### PR DESCRIPTION
**Pull request by hengvenchang**
create student model and view
validate student model
redesgin show and edit of student and teacher page

### CODE REVIEW

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -1 +1,8 @@
- @import "bootstrap";
\ No newline at end of file
+@import 'bootstrap';
+.new-button {
```
Comment (Bunthai) [inline app/assets/stylesheets/application.scss:2]: css class should not be created in here. it should be in another separate file, then be included in the `application.scss`

File: app/views/students/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    </div>
+    <div class="form-group">
+        <%= form.label :email %>
+        <%= form.text_field :email, class: "form-control" %>
+    </div>
+    <div class="form-group">
+              <%= form.label :gender %>
+        <%= form.text_field :gender, class: "form-control" %>
+    </div>
+<div class="field form-group ">
```
Comment (Bunthai) [inline app/views/students/_form.html.erb]: the line should be well-indented

File: app/views/students/_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,46 @@
+<%= form_with(model: student, local: true) do |form| %>
```
Comment (Bunthai) [inline app/views/students/_form.html.erb]: the lines should be well-indented

File: app/views/students/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<h1>Editing Student</h1>
+
+<%= render 'form', student: @student %>
+
+<div class="container text-center">
+<%= link_to 'Back', student_path,class: "btn btn-primary" %> |
```
Comment (Bunthai) [inline app/views/students/edit.html.erb]: the line should be well-indented

File: app/views/students/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,33 @@
+<p id="notice"><%= notice %></p>
```
Comment (Bunthai) [inline app/views/students/index.html.erb]: the lines should be well-indented

File: app/views/students/show.html.erb [hunk 6]
```diff
@@ -0,0 +1,64 @@
+<p id="notice"><%= notice %></p>
```
Comment (Bunthai) [inline app/views/students/show.html.erb]: the lines should be well-indented

**Comment by Bunthai** [comment]
the lines should be well-indented

File: spec/models/student_spec.rb [hunk 7]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Student, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (Bunthai) [inline spec/models/student_spec.rb:4]: should have the spec for validations

File: db/migrate/20190429032902_create_students.rb [hunk 8]
```diff
@@ -0,0 +1,14 @@
+class CreateStudents < ActiveRecord::Migration[5.2]
+  def change
+    create_table :students do |t|
+      t.string :first_name
+      t.string :last_name
+      t.string :phone
+      t.string :email
+      t.integer :gender
```
Comment (Bunthai) [inline db/migrate/20190429032902_create_students.rb:8]: gender is enum, so it should have default value

**Comment by Bunthai** [comment]
specs fail

File: app/views/students/_form.html.erb [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+      <ul>
+      <% student.errors.full_messages.each do |message| %>
+        <li><%= message %></li>
+      <% end %>
+      </ul>
+    </div>
+  <% end %>
+
+  <div class="actions text-center">
+    <%= form.submit%>
```
Comment (Bunthai) [inline app/views/students/_form.html.erb:15]: submit button should be at the bottom

**Comment by Bunthai** [comment]
good and creative UI on student details page

**Comment by thaichors3000** [comment]
late close

**Pull request closed without merging**